### PR TITLE
Refactor `fmtre()`

### PR DESCRIPTION
**Pull request by krader1961**
The current structure of `fmtre()` is hard to understand because it uses
a very unusual pattern. It also causes lint warnings about the final
`break;` at the bottom of the loop. Refactor the function to make the
code easier to understand.

### CODE REVIEW

**Comment by siteshwar** [comment]
Lines changed in this patch are [not covered](http://situ.im/ast/coveragereport/index..._src_lib_libast_string_fmtre.c.html) by tests, can we add tests for them ?

File: src/lib/libast/string/fmtre.c [hunk 1]
```diff
@@ -66,20 +67,24 @@ char *fmtre(const char *as) {
     } else {
         s++;
     }
-    while (1) {
+
+    while (*s) {
         c = *s++;
+        if (c == '*' && !*s) {
+            end = false;
+            break;
```
Comment (siteshwar) [inline src/lib/libast/string/fmtre.c]: Was not this statement breaking out of `switch` statement ?

Comment (krader1961) [reply src/lib/libast/string/fmtre.c]: Yes, but placed in the switch it relied on the `break;` at the bottom of the loop to break out of the loop. It also hid the fact that this was a special-case. Similar to how the `case 0: break;` obscured the fact that the loop is terminated when the end of the string is reached. Which is why I changed `while (1)` to `while (*s)` and dropped that case.

**Comment by krader1961** [comment]
> Lines changed in this patch are not covered by tests, can we add tests for them ?

Actually the coverage is much better than reflected in your report because you built with `-Dbuild-api-tests=false build` which excludes the *src/lib/libast/tests/string/fmtre.c* API test of this function. Having said that the coverage is still unacceptably low so I'll try to improve it.

I think we should remove that option from the directions for producing a coverage report.

**Comment by krader1961** [comment]
I added more tests and we now have 100% line and branch coverage of `fmtre()`. But note I wrote the tests to match the implementation rather than based on the documentation.

**Pull request closed without merging**